### PR TITLE
Read tiny image net

### PR DESCRIPTION
**Pull request by adrianc-a**
See #12 

It might be easier to use the tensorflow dataset api, they have some really nice functions

[see the docs](https://www.tensorflow.org/api_docs/python/tf/data/Dataset)

### CODE REVIEW

**Comment by adrianc-a** [comment]
Let's not merge this until I've gotten reading the test/validation data too.

File: download-data.sh [hunk 1]
```diff
@@ -20,9 +20,9 @@ fi
 
 CIFAR_100_DIRECTORY="${DATA_DIRECTORY}/cifar-100-python"
 if [[ ! -d "$CIFAR_100_DIRECTORY" ]]; then
-    curl https://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz -o "$DATA_DIRECTORY"/cifar-100-batches-py.tar.gz
-    tar -xvf "$DATA_DIRECTORY"/cifar-100-batches-py.tar.gz -C "$DATA_DIRECTORY"
-    rm -rf "$DATA_DIRECTORY"/cifar-100-batches-py.tar.gz
+    curl https://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz -o $DATA_DIRECTORY/cifar-100-batches-py.tar.gz
+    tar -xvf $DATA_DIRECTORY/cifar-100-batches-py.tar.gz -C $DATA_DIRECTORY
```
Comment (leoTiez) [inline download-data.sh]: it is always safer to use the quotes. please put them back in to have it consistent

File: load_data.py [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 import cPickle
 import numpy as np
+from os import environ, path, listdir
+import tensorflow as tf
 
 
-def load_mnist(file_name, dtype, file_path="./data/mnist/"):
+DATA_DIR = './data'
+if 'DATA_DIRECTORY' in environ:
+    DATA_DIR = environ['DATA_DIRECTORY']
```
Comment (leoTiez) [inline load_data.py:11]: Why is this necessary?

**Review by leoTiez — CHANGES_REQUESTED** [review]

File: load_data.py (hunk 2 above)
Comment (adrianc-a) [reply load_data.py:11]: I think it's better not to have to type './data' all the time since a refactor would be more annoying.

Comment (leoTiez) [reply load_data.py:11]: I agree :+1: . But we do not use the environment variable, do we? We always use the string defined at the top. Or am I blind?

Comment (adrianc-a) [reply load_data.py:11]: If you source the script it should export the variable. It's the last line in the script.

Comment (leoTiez) [reply load_data.py:11]: Ah yea, found it. Very sweet I like it.

**Review by leoTiez — APPROVED** [review]

**Review by leoTiez — APPROVED** [review]
